packages/type-utils/src/builtinSymbolLikes.ts

```diff
@@ -69,6 +69,18 @@ export function isReadonlyTypeLike(
     );
   });
 }
+/**
+ * let F = new Function("foo");
+ *      ^ FunctionLike
+ * let I = (callback: Function) => {}
+ *           ^ FunctionLike
+ */
+export function isFunctionLike(program: ts.Program, type: ts.Type): boolean {
+  return (
+    isBuiltinSymbolLike(program, type, 'Function') ||
+    isBuiltinSymbolLike(program, type, 'FunctionConstructor')
```

Comment:
The original rule didn't check for symbols with name `FunctionConstructor`. Should we add tests that contain symbols with name `FunctionConstructor`? But that is probably not within the scope of this pr. IMO we shouldn't change the rule logic in this pr

Reply:
You're right `Function` here has `FunctionConstructor` type. I missed this because in the source code of rule both symbol name and callee identifier's name are compared to `FUNCTION_CONSTRUCTOR`

Then +1 on renaming `isFunctionLike`... Not sure what's the best name for it. Maybe `isFunctionTypeLike` or something like this?

Also we should consider that #8094 most likely will introduce changes to `isBuiltinSymbolLike`.
As per https://github.com/typescript-eslint/typescript-eslint/pull/8094#discussion_r1449120581:
`isBuiltinSymbolLike(program, type, 'Function') || isBuiltinSymbolLike(program, type, 'FunctionConstructor')` recursively visits all subtypes twice, we can optimize to not do the same job twice. #8094 does the following:

```ts
isBuiltinSymbolLike(
  services.program,
  calleeType,
  name => name === 'PromiseConstructor' || name === 'Promise'
)
```

Perhaps we can wait until #8094 is merged and then do the same

Reply:
Yes, sure.

Reply:
> But that is probably not within the scope of this pr. IMO we shouldn't change the rule logic in this pr

Yeah this is just a refactor (`chore`). If you want to change how it works that'd be a separate issue.

👍 I'll mark this as blocked on #8094. Thanks :)

Reply:
#8094 is closed - so we can now pretend it doesn't exist. cc @arka1002, I'm un-blocking this PR. Sorry for the long wait!